src/plugins/plugin-grpc.ts

```diff
@@ -70,7 +70,7 @@ function patchClient(client, api) {
     return function clientMethodTrace() {
       // The span name will be of form "grpc:/[Service]/[MethodName]".
       var span = api.createChildSpan({ name: 'grpc:' + method.path });
-      if (!span) {
+      if (span.type !== api.spanTypes.CHILD) {
```

Comment:
I don't really like this test.

Previously there was a simple way to check if the operation (`createChildSpan`) was succesful. Now we have to check if the operation produced a span of the correct type. This is burdensome for users.

Reply:
This is a check that only needs to be done in plugins, where the expressiveness of distinguishing between untraced and uncorrelated requests can be useful. Most users should never need do to this check.

That being said, it might be nice to have a three-value enum instead with numeric values: `{ REAL = 0, UNTRACED = 1, UNCORRELATED = 2 }`. That way we can have a binary statement to check that spans are non-null: `if (!span.type)` and a comparison if we need to know why it's null: `if (span.type === UNTRACED)` (though obviously the field name should be changed from `type` to something more descriptive). What do you think?

Reply:
I do expect third party authors to write plugins. And the test is still 'wordy'. By reading the test, it is not obvious to me why the test is here. Logically the test the plugin authors want to write is this:

```
if (this request is being sampled) {
  do something
}
```

Instead we see a test that compares the `type` of a span. This doesn't explain what is happening or why comparing the type achieves it.

Perhaps you can add a boolean returning function on the API that would lead to more descriptive tests.

The idea of using integer valued enums with three numeric values but using the fact that `REAL = 0` as a short circuit sounds too hacky to me.